baseplate_test.go

```diff
@@ -145,6 +150,72 @@ func TestServe(t *testing.T) {
 	}
 }
 
+type timestampCloser struct {
+	ts []time.Time
+}
+
+func (c *timestampCloser) Close() error {
+	c.ts = append(c.ts, time.Now())
+	return nil
+}
+
+func TestServeClosers(t *testing.T) {
+	t.Parallel()
+
+	store := newSecretsStore(t)
+	defer store.Close()
+
+	bp := baseplate.NewTestBaseplate(baseplate.Config{StopTimeout: testTimeout}, store)
+
+	pre := &timestampCloser{}
+	post := &timestampCloser{}
+
+	args := baseplate.ServeArgs{
+		Server:       newWaitServer(t, bp, time.Millisecond),
+		PreShutdown:  []io.Closer{pre},
+		PostShutdown: []io.Closer{post},
+	}
+
+	t.Run(
+		"serve closer order",
+		func(t *testing.T) {
+			ch := make(chan error)
+
+			go func() {
+				// Run Serve in a goroutine since it is blocking
+				ch <- baseplate.Serve(
+					context.Background(),
+					args,
+				)
+			}()
+
+			time.Sleep(time.Millisecond)
+			p, err := os.FindProcess(syscall.Getpid())
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			p.Signal(os.Interrupt)
+			err = <-ch
+
+			if len(pre.ts) > 1 {
```

Comment:
shouldn't this be `!=` instead of `>`? you want to fail it if it's not called as well.